turbopack/crates/turbo-rcstr/src/lib.rs

```diff
@@ -256,7 +259,15 @@ impl Default for RcStr {
 
 impl PartialEq for RcStr {
     fn eq(&self, other: &Self) -> bool {
-        self.as_str() == other.as_str()
+        match (self.tag(), other.tag()) {
+            (DYNAMIC_TAG, DYNAMIC_TAG) => {
+                let l = unsafe { deref_from(self.unsafe_data) };
+                let r = unsafe { deref_from(other.unsafe_data) };
+                l.hash == r.hash && l.value == r.value
```

Comment:
This could be `l == r` if you swap the fields in `PrehashedString` and put `hash` first.